models/dvmdostem/R/write.config.dvmdostem.R

```diff
@@ -0,0 +1,353 @@
+  local_rundir <- file.path(settings$rundir, run.id)    # on local machine for staging
+  rundir <- file.path(settings$host$rundir, run.id)     # on remote machine for execution
+  outdir <- file.path(settings$host$outdir, run.id)
+  appbinary <- settings$model$binary
+  appbinary_path <- dirname(appbinary)                  # path of dvmdostem binary file
+
+  # Copy the base set of dvmdostem parameters and configurations into the
+  # run directory. Some of the values in these files will be overwritten in
+  # subsequent steps, but copying everything up makes sure that all the
+  # necessary files exist for a tem run - the config file and the parameter
+  # files in this case.
+  if (dir.exists(file.path(rundir, 'config'))) {
+    unlink(file.path(rundir, 'config'), recursive=TRUE)
+  }
+  system2(paste0("cp"),
+          wait=TRUE,
+          args=(c("-r",
+                  file.path(appbinary_path, 'config'),
+                  file.path(rundir, 'config'))))
+
+  if (dir.exists(file.path(rundir, 'parameters'))) {
+    unlink(file.path(rundir, 'parameters'), recursive=TRUE)
+  }
+  system2(paste0("cp"),
+          wait=TRUE,
+          args=(c("-r",
+                  file.path(appbinary_path, 'parameters'),
+                  file.path(rundir, 'parameters'))))
+
+  # (1)
+  # Read in a parameter data block from dvmdostem
+
+  # Build a dataframe from the incoming trait.values. trait.values
+  # contains the meta-analysis posteriors values for each parameter (trait)
+  # that we are hoping to "inject" into dvmdostem. Convert to a dataframe
+  # for easier indexing...
+  trait_df <- convert.samples.dvmdostem(trait.values)
+  ## !! TODO: Should run this as below so that only trait names remain !!
+  ## !! But if we do that now, the parsing below wont work.  Need to
+  ## !! separate parsing of PFT name for json and trait conversion for BETYdb
+  #trait_df <- convert.samples.dvmdostem(trait.samples = trait.values[[settings$pfts$pft$name]])
+  
+  # Now we have to read the approporate values out of the trait_df
+  # and get those values written into the parameter file(s) that dvmdostem will
+  # need when running. Because the dvmdostem parameters have a sort of
+  # interesting, semi-standardized, space delimited, column format, we'll
+  # use some helper scripts from dvmdostem that allow us to more easily
+  # handle the parameter files and write our new trait values into the correct
+  # place. The basic flow will be like this:
+  #  - Read dvmdostem parameter file into json object, load into memory
+  #  - Update the in-memory json object
+  #  - Write the json object back out to a new dvmdostem parameter file
+
+  # Start by figuring out which community number we are working on.
+  # This is important because we need to pull the correct "community block"
+  # from the dvmdostem parameter files. Each trait in the traits dataframe
+  # should have a name like this:
+  #   CMT04.Salix.extinction_coefficient_diffuse
```

Comment:
not sure how/why this would be true. If "CMT04.Salix" is your pft and "extinction_coefficient_diffuse" is your trait, then one of these would be the row and the other would be a column. Or, if you leave traits in a list (recommended), the PFT will be the elements of the list and the parameters will be the named elements in the vectors.